### ISSUE TITLE
docker exec -d often just stops working

### ISSUE THREAD

**Issue by corrieb** [vmware/vic#4598]
I've observed two problems with docker exec -d in the current builds.

(1) Some commands simply don't work. They *should* work and I've confirmed that in Caglar's PR #4339 they do work with those fixes.

(2) If you issue a couple of execs that don't work, exec simply stops working completely. Subsequent calls which previously worked cease working. It's therefore hard to know if you got a command wrong, if exec isn't working because of (1) or if it's got into state (2).

I'll attach details on how to reproduce both scenarios.

**Comment by corrieb** [comment]
Here's a run-through of how to reproduce, and I'll attach the debug logs from this very run I'm about to do. These require a container that runs sshd, so that you can get in and look at the side effects of whether an exec worked or not. I include instructions that use one from my own Docker Hub repo.

Running with build 9461 (the latest available today).

(1) Let's start with a case where exec works:

```
docker run --name test -d --net=ExternalNetwork bensdoings/dind-centos-ssh
docker exec -d test /bin/touch /tmp/foo
docker inspect --format "{{ .NetworkSettings.Networks.ExternalNetwork.IPAddress }}" test
ssh vmware@<ip> (pwd vmware)
ls /tmp            # foo exists
```
In this example, we should see that /tmp/foo exists. Exec worked!

(2) Let's now create a scenario where exec doesn't work. Still logged into the same cVM, create a script, log out and get exec to run it.

```
sudo bash
echo 'echo $1 > /tmp/bar' > /bin/echotest
chmod 777 /bin/echotest
/bin/echotest hello
cat /tmp/bar       # should see "hello" in the file
rm /tmp/bar
# exit back to shell with VIC docker client

docker exec -d test /bin/echotest fromDocker
ssh vmware@<ip>
cat /tmp/bar       # file does not exist
```

(3) Now run the failing docker exec command from (2) 3 times and finally try to do the "touch" command that worked from (1). It no longer works.

```
docker exec -d test /bin/echotest fromDocker
docker exec -d test /bin/echotest fromDocker
docker exec -d test /bin/echotest fromDocker
docker exec -d test /bin/touch /tmp/abc
ssh vmware@<ip>
ls /tmp            # abc does not exist
```

**Comment by corrieb** [comment]
[container-logs (5).zip](https://github.com/vmware/vic/files/900970/container-logs.5.zip)

**Comment by hickeng** [comment]
I've been unable to recreate this once the interpreter is specified at the head of the script:
```
/tmp # echo -e '#!/bin/ash\ndate > /tmp/date\necho $1 >> /tmp/date' > /bin/echotest
/tmp # cat /bin/echotest
#!/bin/ash
date > /tmp/date
echo $1 >> /tmp/date
/tmp # ls -l /tmp
total 0
/tmp #
```
```
$ docker exec -d b /bin/echotest hello
```
```
/tmp # ls -l /tmp
total 4
-rw-rw-rw-    1 root     root            35 Apr  6 12:59 date
/tmp # cat /tmp/date
Thu Apr  6 12:59:23 UTC 2017
hello
/tmp #
```

We can and *should* be returning the `exec format error` but I don't think this is a failure to launch.

**Comment by hickeng** [comment]
It looks like we're dealing with the error return from the PL in #4339 but we could cherry pick the following:
https://github.com/vmware/vic/pull/4339/files#diff-a120d0dc93ea756095ab5dc53ebd5d82

However we're still not propagating that in the personality ( @caglar10ur )

**Comment by corrieb** [comment]
Confirmed George's findings. 

Confirmed that I can do what I need with RC2 and DDC. @hickeng should we close this, since all this should be addressed by @caglar10ur exec PR?

**Comment by hickeng** [comment]
@corrieb I've added this issue to the set that are addressed by #4339 - it should be closed automatically when that PR merges.
